### PR TITLE
chore: multistage build and confugirable GOARCH

### PR DESCRIPTION
**Pull request by kesoji**
**CAUTION: this PR depends on #1**

## 1. Multistage build for javascript compilation

There is an old version node in golang:1.19 so the below warning was showing
image

Changed to use node:16 for javascript compilation

## 2. Make GOARCH settable

`make release` in M1 Mac caused `exec format error` when starting.

Changed Dockerfile to receive a GOARCH build arg.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,15 +1,21 @@
...
 build:
 	cd cmd/catfish/static && npm ci && npm run build
-	${CMDS} | xargs -I@ go build -ldflags "-X ${CONFIG_PKG}.AppVersion=${VERSION}" -o bin/@ ./cmd/@
+	${CMDS} | xargs -I@ go build -ldflags "-X ${CONFIG_PKG}.AppVersion=${VERSION}" -buildvcs=${BUILDVCS} -o bin/@ ./cmd/@
 
-release:
+release: release-js release-app
+
+release-app:
+	${CMDS} | CGO_ENABLED=0 GOOS=linux GOARCH=${GOARCH} xargs -I@ \
```
Comment (soranoba) [inline Makefile:14]: Ah... I'm sorry. Will it be successful to build when deleting the GOARCH and GOOS?

```
build: build-js build-app

build-app:
    ${CMDS} | xargs -I@ go build -ldflags "-X ${CONFIG_PKG}.AppVersion=${VERSION}" -o bin/@ ./cmd/@

build-js:
    cd cmd/catfish/static && npm ci && npm run build
```

I thought this one would be better. Does it work as you expect?

File: Dockerfile [hunk 2]
```diff
@@ -1,15 +1,27 @@
...
+WORKDIR /app
+COPY ./Makefile ./
+RUN make release-js
 
-RUN apt-get update && \
-    apt-get install -y npm && \
-    apt-get purge -y
+#==========================
+
+FROM golang:1.19 AS builder
```
Comment (soranoba) [inline Dockerfile:14]: Please change builder to builder-app

File: Makefile (hunk 1 above)
Comment (kesoji) [reply Makefile]: @soranoba it's enough. Deleted GOOS and GOARCH.

File: Dockerfile (hunk 2 above)
Comment (kesoji) [reply Dockerfile]: @soranoba Changed

File: Makefile [hunk 3]
```diff
@@ -2,14 +2,16 @@ VERSION     := $(shell git describe --always --tags --abbrev=10)
...
 	${CMDS} | xargs -I@ go build -ldflags "-X ${CONFIG_PKG}.AppVersion=${VERSION}" -o bin/@ ./cmd/@
 
-release:
+build-js:
 	cd cmd/catfish/static && npm ci && npm run build
-	${CMDS} | CGO_ENABLED=0 GOOS=linux GOARCH=amd64 xargs -I@ \
-		go build -ldflags "-s -w -X ${CONFIG_PKG}.AppVersion=${VERSION}" -o bin/@ -a ./cmd/@
+
+release-app:
+	${CMDS} | xargs -I@ go build -ldflags "-s -w -X ${CONFIG_PKG}.AppVersion=${VERSION}" -o bin/@ -a ./cmd/@
```
[the remark this answers is not in the document]
Comment (soranoba) [reply Makefile:14]: All right!

**Review by soranoba — APPROVED** [review]
thx!

**Pull request merged**